MediaBrowser.Controller/ClientEvent/ClientEventLogger.cs

```diff
@@ -23,8 +23,12 @@ public async Task<string> WriteDocumentAsync(string clientName, string clientVer
         {
             var fileName = $"upload_{clientName}_{clientVersion}_{DateTime.UtcNow:yyyyMMddHHmmss}_{Guid.NewGuid():N}.log";
             var logFilePath = Path.Combine(_applicationPaths.LogDirectoryPath, fileName);
-            await using var fileStream = new FileStream(logFilePath, FileMode.CreateNew, FileAccess.Write, FileShare.None);
-            await fileContents.CopyToAsync(fileStream).ConfigureAwait(false);
```

Comment:
This seems to be a common anti-pattern... the `using` line should always have braces enclosing the body using the `IDisposable` object, that's the entire purpose OF the using statement. Seems that when used here, we're seeing a false understanding that the next line is the scope of the using, rather it is "anytime you want, but before the next-outer scope close" is the real spec.

I think we need to make a sweep looking for this (I just fixed an issue caused by this error in PR #9559), and I might be the right person to do it :)

Reply:
This isn't an anti-pattern but was a feature added in C# 8.
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/using

Reply:
It does cause the class of bugs I fixed in PR #9559.

Reply:
Looks like some good reading here for me.